src/components/BodyWeight/WeightFilter/WeightFilter.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { Button, ButtonGroup, Box } from '@mui/material';
+
+interface WeightFilterProps {
+  onFilter: (filter: string) => void;
+}
+
+const WeightFilter: React.FC<WeightFilterProps> = ({ onFilter }) => {
+  const [activeFilter, setActiveFilter] = useState('all');
+
+  const handleFilterClick = (filter: string) => {
+    setActiveFilter(filter);
+    onFilter(filter);
+  };
+
+  return (
+    <Box sx={{ display: 'flex', justifyContent: 'center', my: 2 }}>
+      <ButtonGroup variant="outlined" aria-label="outlined primary button group">
+        <Button 
+          variant={activeFilter === 'all' ? 'contained' : 'outlined'} 
+          onClick={() => handleFilterClick('all')}
+        >
+          All
```

Comment:
can you mark the texts as translatable? you just need to add a key to public/locales/en/translation.json and then you can use it with `t('')`